R/looping.R

```diff
@@ -55,65 +51,61 @@ slide_forecaster <- function(data,
     net_slide_training <- slide_training + n_training_pad
   }
   # restrict the dataset to areas where training is possible
-  start_date <- min(archive$DT$time_value) + net_slide_training
-  end_date <- max(archive$DT$time_value) - forecaster_args$ahead
+  start_date <- min(epi_archive$DT$time_value) + net_slide_training
+  end_date <- max(epi_archive$DT$time_value) - forecaster_args$ahead
   valid_predict_dates <- seq.Date(from = start_date, to = end_date, by = date_range_step_size)
 
   # first generate the forecasts
   before <- n_training + n_training_pad - 1
-  ## TODO: epix_slide doesn't support infinite `before`
-  ## https://github.com/cmu-delphi/epiprocess/issues/219
-  if (before == Inf) before <- 365L * 10000
-  res <- epix_slide(archive,
-    function(data, gk, rtv, ...) {
-      # TODO: Can we get rid of this tryCatch and instead hook it up to targets
-      #       error handling or something else?
-      #       https://github.com/cmu-delphi/exploration-tooling/issues/41
-      tryCatch(
-        {
-          do.call(
-            forecaster,
-            append(
-              list(
-                epi_data = data,
-                outcome = outcome,
-                extra_sources = extra_sources
-              ),
-              forecaster_args
-            )
-          )
-        },
-        error = function(e) {
-          if (interactive()) {
-            browser()
-          } else {
-            dump_vars <- list(
-              data = data,
-              rtv = rtv,
-              forecaster = forecaster,
-              forecaster_args = forecaster_args,
-              e = e
-            )
-            saveRDS(dump_vars, "slide_forecaster_error.rds")
-            e
-          }
-        }
-      )
-    },
-    before = before,
-    ref_time_values = valid_predict_dates,
+  forecaster_args <- rlang::dots_list(
+    !!!list(
+      outcome = outcome,
+      extra_sources = extra_sources
+    ),
+    !!!forecaster_args,
+    .homonyms = "last"
+  )
+  forecaster_wrapper <- function(x) {
+    inject(forecaster(epi_data = x, !!!forecaster_args))
+  }
+  res <- epix_slide_simple(
+    epi_archive,
+    forecaster_wrapper,
+    valid_predict_dates,
+    before,
+    cache_key = cache_key
   )
-  res %<>% select(-time_value)
-  names(res) <- sub("^slide_value_", "", names(res))
 
-  # append the truth data
-  true_value <- archive %>%
-    epiprocess::epix_as_of(archive$versions_end) %>%
+  true_value <- epi_archive %>%
+    epiprocess::epix_as_of(epi_archive$versions_end) %>%
     select(geo_value, time_value, !!outcome) %>%
     rename(true_value = !!outcome)
-  res %<>%
+  res %>%
     inner_join(true_value,
       by = join_by(geo_value, target_end_date == time_value)
     )
-  return(res)
+}
+
+epix_slide_simple <- function(epi_archive, forecaster, ref_time_values, before, cache_key = NULL) {
```

Comment:
just confirming, it looks like the `cache_key` is currently only used by `forecaster_profiling.R`?

Reply:
It's also in make_forecasts_and_scores in targets-exploration-common.R.